src/components/_global/BalTooltip/BalTooltip.vue

```diff
@@ -83,22 +88,105 @@ onMounted(() => {
       <BalIcon :name="iconName" :size="iconSize" :class="iconClass" />
     </slot>
   </button>
-  <div
-    ref="content"
-    class="tooltip text-xs text-black dark:text-white bg-white dark:bg-gray-800 font-medium shadow rounded-md border dark:border-gray-900 z-50"
-    :class="tooltipClasses"
-    v-bind="$attrs"
-  >
-    <p v-if="text" v-text="text" />
-    <slot v-else />
+  <div ref="content" class="tooltip" :class="tooltipClasses" v-bind="$attrs">
+    <div :class="tooltipPad" class="tooltip-content">
+      <p class="tooltip-text" v-if="text" v-text="text" />
+      <slot v-else />
+    </div>
   </div>
 </template>
 <style>
 .tooltip {
-  display: none;
+  @apply z-50 hidden relative shadow-sm;
+}
+
+.dark .tooltip {
+  @apply shadow-none;
+}
+
+/* Light mode gray shadow */
+.tooltip:before {
+  background-blend-mode: soft-light, soft-light, normal;
+  background: radial-gradient(
+    ellipse at center,
+    rgba(0, 0, 0, 0.6),
+    transparent
+  );
+  content: '';
+  width: 100%;
+  height: 100%;
+  display: block;
+  position: absolute;
+  top: 0;
+  left: 0;
+  filter: blur(40px);
+  z-index: -1;
+  opacity: 0;
+  animation: fadeIn 0.4s ease-out 0.05s both;
+}
+
+/* Dark mode radial gradient shadow */
+.dark .tooltip:before {
+  background-blend-mode: soft-light, soft-light, normal;
+  background: radial-gradient(ellipse at left, yellow, transparent),
+    radial-gradient(ellipse at bottom right, blue, transparent),
+    radial-gradient(ellipse at top, red, transparent);
+  content: '';
 }
 
 .tooltip[data-show] {
   display: block;
 }
+
+.tooltip[data-popper-placement='top'] .tooltip-content {
+  opacity: 0;
+  animation: fadeInMoveUp 0.2s ease-out both;
+}
+
+.tooltip[data-popper-placement='bottom'] .tooltip-content {
+  opacity: 0;
+  animation: fadeInMoveDown 0.2s ease-out both;
+}
+
+/* bottom triangle for top placed tooltips */
+.tooltip[data-popper-placement='top'] .tooltip-content:before {
+  border-top: solid #fff 8px;
+  border-left: solid transparent 8px;
+  border-right: solid transparent 8px;
```

Comment:
I'm a little concerned with all the custom sizing and hex colors here and below. Is it difficult to use to tailwind classes here?

Reply:
Ahh ok, I didn't realise that was not possible in v2 tbh.